sensor/common/compliance/nodescan_handler_impl.go

```diff
@@ -6,35 +6,51 @@ import (
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/centralsensor"
 	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
 )
 
 type nodeScanHandlerImpl struct {
 	nodeScans <-chan *storage.NodeScanV2
-	toCentral chan *central.MsgFromSensor
+	toCentral <-chan *central.MsgFromSensor
+	// toCentralMux prevents the race condition between Start() [writer] and ResponsesC() [reader]
```

Comment:
From what I can tell, we call `Start()` for each component before eventually calling `ResponsesC()` for each (see https://github.com/stackrox/stackrox/blob/3.72.2/sensor/common/sensor/sensor.go#L210. it looks like we call `Start()` for each component and then we eventually run `ProcessMessage()` and `ResponsesC()` concurrently). It definitely does not hurt at all to be defensive, but I don't think other sensor components have bothered.

Reply:
Similarly, I wonder if we should just have some flag which indicates if `Start()` has been called and only allow `ResponsesC` continue once that happens? Or we even just comment that you must call `ResponsesC` *after* `Start()` completes and leave it up to the caller

Reply:
I know that some usage patterns are not used in our setup, but my tests reported race or a leak and I wanted them to be clean. Moreover, we discussed the restartablilty of sensor components in the F2F meeting and we may be soon rely on it.